# Projet Fil Rouge Centrale #

Victor Bonlieu-Fortier, Charles Nikdel

---

## Importation librairies ##

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import utils as ft_nous


In [2]:
rename_columns = [
					"index",
					"date_urg",
					"conduite",
					"m_entry",
					"m_venue",
					"age",
					"atcd_med",
					"atcd_chir",
					"cst_paspad",
					"fc_ad",
					"tmp_ad",
					"saoxy_ad",
					"fr_ad",
					"ccmu",
					"salle",
					"date_box",
					"anamn",
					"obs",
					"have_bio",
					"have_radio",
					"have_echo",
					"have_scan",
					"have_irm",
					"have_specia",
					"date_prescri_bio",
					"date_prelev_bio",
					"code_diag",
					"intitule_diag",
					"cac_admi",
					"cac_sejour",
					"date_sortie",
					"date_deb_pec_med",
					"dossier_hopit",
					"type_orient",
					"destination",
					"transfert_service",
					"transfert_hopit",
					"cp_ville",
					"ville"
					]
rename_columns


['index',
 'date_urg',
 'conduite',
 'm_entry',
 'm_venue',
 'age',
 'atcd_med',
 'atcd_chir',
 'cst_paspad',
 'fc_ad',
 'tmp_ad',
 'saoxy_ad',
 'fr_ad',
 'ccmu',
 'salle',
 'date_box',
 'anamn',
 'obs',
 'have_bio',
 'have_radio',
 'have_echo',
 'have_scan',
 'have_irm',
 'have_specia',
 'date_prescri_bio',
 'date_prelev_bio',
 'code_diag',
 'intitule_diag',
 'cac_admi',
 'cac_sejour',
 'date_sortie',
 'date_deb_pec_med',
 'dossier_hopit',
 'type_orient',
 'destination',
 'transfert_service',
 'transfert_hopit',
 'cp_ville',
 'ville']

In [3]:
df = pd.read_excel("../database/db_urg_liteplus.xlsx", index_col=0, header=0, names = rename_columns)
df


date_urg           conduite  \
index                                          
1     2016-06-01 00:16:00   AMBULANCE PRIVEE   
2     2016-06-01 00:18:00   AMBULANCE PRIVEE   
3     2016-06-01 00:36:00           POMPIERS   
4     2016-06-01 00:43:00  MOYENS PERSONNELS   
5     2016-06-01 01:16:00   AMBULANCE PRIVEE   
...                   ...                ...   
61488 2017-02-26 02:34:00          SAMU-SMUR   
61489 2017-02-26 02:43:00  MOYENS PERSONNELS   
61490 2017-02-26 02:52:00  MOYENS PERSONNELS   
61491 2017-02-26 03:19:00  MOYENS PERSONNELS   
61492 2017-02-26 03:26:00  MOYENS PERSONNELS   

                                                 m_entry  \
index                                                      
1      Malaise, difficulté respi, palpitation, Pas de...   
2      AEG, Hyperthermie et difficulté respi en maiso...   
3                                                    NaN   
4      1A, Hyperthermie ne cédant pas au paracetamol,...   
5      Douleur bras gauche et mollet gauche, hémiplég...   
...                                                  ...   
61488                                                NaN   
61489                                                NaN   
61490                       suspicion cellulite dentaire   
61491                                  intox alimentaire   
61492                  céphalées depuis 1 mois, stat neg   

                                                 m_venue  age  \
index                                                           
1                      Cardiologie: Palpitations Malaise   33   
2                    Maladies infectieuses: Hyperthermie   78   
3                                                    NaN   77   
4                    Maladies infectieuses: Hyperthermie   38   
5                                  Cardiologie: Phlébite   69   
...                                                  ...  ...   
61488                                                NaN   78   
61489                                                NaN   36   
61490            Stomatologie: Abcès/Cellulite dentaire    45   
61491  Gastro-entérologie: Douleur abdominale non épi...   18   
61492                              Neurologie: Céphalées   86   

                                                atcd_med       atcd_chir  \
index                                                                      
1                                                    NaN             NaN   
2                                                    NaN             NaN   
3                                                    NaN             NaN   
4                                                    NaN             NaN   
5      décembre 2007 : infarctus capsulo caudé droit ...             NaN   
...                                                  ...             ...   
61488                                                NaN             NaN   
61489                                                NaN             NaN   
61490            hypertriglycéridémie  crise de goutte    amygdalectomie   
61491                                                NaN             NaN   
61492                           hypertrophie de prostate               0   

      cst_paspad fc_ad tmp_ad  ... cac_sejour   date_sortie date_deb_pec_med  \
index                          ...                                             
1         110/56    52   34.9  ...     3093.0  42522.176389     42522.043750   
2          90/41    77   36.3  ...     3102.0  42522.461806     42522.093056   
3         137/77    97   36.5  ...     3102.0  42522.127778              NaN   
4         117/71   118   38.2  ...     3093.0  42522.201389     42522.081250   
5         137/79    70   36.1  ...     3102.0  42522.663194     42522.107639   
...          ...   ...    ...  ...        ...           ...              ...   
61488      44058    72   36.3  ...     3102.0  42792.137500     42792.108333   
61489        NaN   NaN    NaN  ...     3098.0  42792.122

In [5]:
df.info()


<class 'pandas.core.frame.DataFrame'>
Index: 171708 entries, 1 to 171708
Data columns (total 38 columns):
 #   Column             Non-Null Count   Dtype         
---  ------             --------------   -----         
 0   date_urg           171708 non-null  datetime64[ns]
 1   conduite           170977 non-null  object        
 2   m_entry            159960 non-null  object        
 3   m_venue            157682 non-null  object        
 4   age                171708 non-null  int64         
 5   atcd_med           39248 non-null   object        
 6   atcd_chir          26563 non-null   object        
 7   cst_paspad         121488 non-null  object        
 8   fc_ad              122162 non-null  object        
 9   tmp_ad             126404 non-null  object        
 10  saoxy_ad           108214 non-null  object        
 11  fr_ad              4429 non-null    object        
 12  ccmu               158213 non-null  object        
 13  salle              171708 non-null  object       

On voit que bon, y a beaucoup de colonne mais pas mal qui contienne rien d'interessant, on va donc les supprimer
Dans l'ordre ceux que je pense supprimer et concatenner pour certaines : 

- obs sert pas a grand chose et y en a pas bcq, c'est les observations des inf genre EVA ou autre
- code diag donne le code du diagnostic, on a deja le libelle donc pas besoin
- cac admi jsp vraiment ce que c'est ca doit etre un code de protocole, mais il faut juste récup le numéro comme le cac_sejour
- dossier_hopit y es casi nul donc on peut tej
- type orientation, c'est destination mais moins précis, on peut tej
- transfert service c'est destination mais juste pour les services internes, on peut tej
- transfert etab c'est destination mais juste pour les autres etablissements, y en as pas enorme, je pense on peut tej et et juste modif pour dire autre etab
- le code postal et la ville jsp si c'est utile la vérité à voir, je pense qu'on peut retirer le code postal et garder la ville pour des vizu de zinzin
- frequence respi y en as pas bcq on peut retirer
- Je pense qu'on peut retirer aussi tout ceux qui n'ont pas un code ccmu, ca veut dire ceux qui ont pas été triagé et donc pas pris en charge par les urgences
- salle enfaite c'est une liste de leur voyage dans les urgences, jsp si c'est utile non plus , ca clc à parser et jsp si ca sera utile

Pour réduire le nbr de colonne et le contenu en info :

- Conduite, on peut changer ca en 1 2 3 4 5 ... y 'en a pas bcq
- motif entrée et admission : clean et tokenizer comme on as fait sur twitter
- atcd med et chir idem 
- les constantes jsp pq aucune est en int il faut convert : trouver un truc pck la pression sanguine en 110 / 56 ca clc par exemple vu que c'est pas un int
- convertir la température en int, pour remplacer les NaN juste mettre la température normale
- la saturation en oxy, pareil remplacer les NaN par la saturation normale et convertir en int sachant que y a des inf clc qui ont met AA (air ambiant) du coup les retirer
- anamnèse, on peut tokenizer et clean comme on as fait sur twitter
- tout les have, à transformer en 0 1
- les date prescri bio et prelev, peut être juste faire un delta entre les deux, y en as ils sont pas prescri donc on peut mettre 0 et d'autre qui sont prescri mais pas prelevé donc mettre la date de prescri ou la moyenne des dates de prescri jpense
- intitule diag à tokenizer et clean
- parser destination pour juste garder le service / hopital / autre hopital
- ville ca a l'air d'être bon

In [42]:
df.columns


Index(['date_urg', 'conduite', 'm_entry', 'm_venue', 'age', 'atcd_med',
       'atcd_chir', 'cst_paspad', 'fc_ad', 'tmp_ad', 'saoxy_ad', 'fr_ad',
       'ccmu', 'salle', 'date_box', 'anamn', 'obs', 'have_bio', 'have_radio',
       'have_echo', 'have_scan', 'have_irm', 'have_specia', 'date_prescri_bio',
       'date_prelev_bio', 'code_diag', 'intitule_diag', 'cac_admi',
       'cac_sejour', 'date_sortie', 'date_deb_pec_med', 'dossier_hopit',
       'type_orient', 'destination', 'transfert_service', 'transfert_hopit',
       'cp_ville', 'ville'],
      dtype='object')

conduite
1    36612
2    14036
3     5836
5     1023
4      617
6      467
7      144
8       67
9       59
Name: count, dtype: int64

In [4]:
df = df.dropna(subset=["ccmu"])
df = df.drop(["obs","anamn","salle","m_entry","atcd_chir", "atcd_med","code_diag","dossier_hopit","type_orient","transfert_service","transfert_hopit","fr_ad","ville"],axis=1)
df.info()


<class 'pandas.core.frame.DataFrame'>
Index: 58861 entries, 1 to 61492
Data columns (total 26 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   date_urg          58861 non-null  datetime64[ns]
 1   conduite          58828 non-null  object        
 2   m_venue           55698 non-null  object        
 3   age               58861 non-null  int64         
 4   cst_paspad        40002 non-null  object        
 5   fc_ad             40152 non-null  object        
 6   tmp_ad            40937 non-null  object        
 7   saoxy_ad          34581 non-null  object        
 8   ccmu              58861 non-null  object        
 9   date_box          34251 non-null  float64       
 10  anamn             47508 non-null  object        
 11  have_bio          58861 non-null  object        
 12  have_radio        58861 non-null  object        
 13  have_echo         58861 non-null  object        
 14  have_scan         58861 non

In [5]:
df = ft_nous.lstdate(df)
df = ft_nous.convertir_conduite(df,"conduite")
df = ft_nous.pipelinett(df)
df.head()



date_urg  conduite                              m_venue  age  \
index                                                                           
1     2016-06-01 00:16:00         6    Cardiologie: Palpitations Malaise   33   
2     2016-06-01 00:18:00         6  Maladies infectieuses: Hyperthermie   78   
3     2016-06-01 00:36:00         6                                  NaN   77   
4     2016-06-01 00:43:00         6  Maladies infectieuses: Hyperthermie   38   
5     2016-06-01 01:16:00         6                Cardiologie: Phlébite   69   

      fc_ad  tmp_ad  saoxy_ad ccmu                   date_box  \
index                                                           
1        52    34.9        97    1 2016-06-01 01:05:59.999997   
2        77    36.3        98    2 2016-06-01 01:32:00.000001   
3        97    36.5        96    2 2016-06-01 00:40:59.999997   
4       118    38.2        98    1 2016-06-01 01:36:00.000003   
5        70    36.1        97    2 2016-06-01 01:36:00.000003   

                                                   anamn  ...  \
index                                                     ...   
1      MDC: Dyspnée aiguë et palpitations    ATCDS:  ...  ...   
2      vient pour hyperthermie    ATCD:  DID  retard ...  ...   
3                                                    NaN  ...   
4      patient de 38 ans appelant le SAMU pour épisod...  ...   
5      Patiente de 69 ans amenée pour douleur membre ...  ...   

                 date_prelev_bio  \
index                              
1     2016-06-01 01:34:59.999997   
2     2016-06-01 01:34:00.000002   
3     1899-12-30 00:00:00.000000   
4     2016-06-01 01:39:59.999996   
5     2016-06-01 01:34:59.999997   

                                           intitule_diag cac_admi cac_sejour  \
index                                                                          
1                                                Dyspnée     3093     3093.0   
2                                 Fièvre, sans précision     3093     3102.0   
3                                Dysarthrie et anarthrie     3093     3102.0   
4      Autres affections précisées de la peau et du t...     3093     3093.0   
5                   Douleur au niveau d'un membre - Bras     3093     3102.0   

                     date_sortie           date_deb_pec_med destination  \
index                                                                     
1     2016-06-01 04:14:00.000001 2016-06-01 01:02:59.999999           1   
2     2016-06-01 11:05:00.000003 2016-06-01 02:13:59.999995           4   
3     2016-06-01 03:04:00.000002 1899-12-30 00:00:00.000000        3281   
4     2016-06-01 04:50:00.000000 2016-06-01 01:57:00.000000           1   
5     2016-06-01 15:54:59.999996 2016-06-01 02:35:00.000000           1   

      cp_ville  systo diasto  
index                         
1      59139.0  110.0   56.0  
2      59310.0   90.0   41.0  
3      59480.0  137.0   77.0  
4      59136.0  117.0   71.0  
5      59130.0  137.0   79.0  

[5 rows x 27 columns]

'AMB PUBLIQUE CHRU '

Maintenant je vais faire des fonctions pour traiter les données et les remettre dans le bon type en général

ValueError: invalid literal for int() with base 10: 'AMB PUBLIQUE CHRU '

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 58861 entries, 1 to 61492
Data columns (total 27 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   date_urg          58861 non-null  datetime64[ns]
 1   conduite          58861 non-null  int64         
 2   m_venue           55698 non-null  object        
 3   age               58861 non-null  int64         
 4   fc_ad             40152 non-null  object        
 5   tmp_ad            40937 non-null  float64       
 6   saoxy_ad          58861 non-null  int64         
 7   ccmu              58861 non-null  object        
 8   date_box          58861 non-null  datetime64[ns]
 9   anamn             47508 non-null  object        
 10  have_bio          58861 non-null  object        
 11  have_radio        58861 non-null  object        
 12  have_echo         58861 non-null  object        
 13  have_scan         58861 non-null  object        
 14  have_irm          58861 non